### PR TITLE
INTMDB-527-Feature private endpoint regional mode

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by rdharmedran — APPROVED** [review]
LGTM

**Comment by Zuhairahmed** [comment]
Please resolve branch conflicts

**Comment by Zuhairahmed** [comment]
Missing GitHub Build + Linting Error checks, please add

File: cfn-resources/private-endpoint-regional-mode/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+cd ${repo_root}/cfn-resources/private-endpoint-regional-mode
+./test/private-endpoint-regional-mode.create-sample-cfn-request.sh > test.request.json
+echo "Sample request:"
+cat test.request.json
+cfn invoke CREATE test.request.json
+cfn invoke DELETE test.request.json
+```
+
+Both CREATE & DELETE tests must pass.
+
```
Comment (Zuhairahmed) [inline cfn-resources/private-endpoint-regional-mode/README.md:35]: Missing Example section, please add and link to Examples folder: https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples

File: examples/private-endpoint-regional-mode/privateEndpointRegionalMode.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  "AWSTemplateFormatVersion" : "2010-09-09",
+  "Description" : "This template creates a Project on the MongoDB Atlas API, this will be billed to your Atlas account.",
```
Comment (Zuhairahmed) [inline examples/private-endpoint-regional-mode/privateEndpointRegionalMode.json:3]: This appears to be for Project resource. Please update for Private Endpoint Regional Mode

**Review by Zuhairahmed — CHANGES_REQUESTED** [review]
Thanks @govardhanpagidi! Few comments if you can please address and then ping in Slack to review again when ready. Otherwise great work!

File: cfn-resources/private-endpoint-regional-mode/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+## CloudFormation Examples
+
+Please see the [CFN Template](test/private-endpoint-regional-mode.sample-cfn-request.json) for example resource
```
Comment (Zuhairahmed) [inline cfn-resources/private-endpoint-regional-mode/README.md:38]:
```suggestion
Please see the [Private Endpoint Regional Mode](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples/private-endpoint-regional-mode) for example resource
```

**Review by Zuhairahmed — APPROVED** [review]
@govardhanpagidi few more minor doc comments, please review. Otherwise not blocking, LGTM

**Comment by Zuhairahmed** [comment]
Also please resolve branch conflicts + Missing GitHub Build + Linting Error checks, please add

**Review by ParthasarathyVaradhan — APPROVED** [review]

**Review by matt-condon — APPROVED** [review]

**Pull request merged**